### PR TITLE
Docs: Split Traffic Analytics to a How-to guide and a Feature entry (Diátaxis)

### PR DESCRIPTION
**Pull request by benjaoming**
Refs: #9746 

---
:books: Documentation previews :books:

- User's documentation (`docs`): https://docs--9677.org.readthedocs.build/en/9677/

- Developer's documentation (`dev`): https://dev--9677.org.readthedocs.build/en/9677/

### CODE REVIEW

File: docs/user/analytics.rst
Comment (humitos) [inline docs/user/analytics.rst:4]: Just wanted to mention that this guide also explains "how to _remove_ Google Analytics"

Comment (benjaoming) [reply docs/user/analytics.rst]: Good idea! Fixed!

File: docs/user/guides/administrators.rst
Comment (ericholscher) [inline docs/user/guides/administrators.rst:16]: This is an odd title, and again we should just use the article title directly?

File: docs/user/analytics.rst
Comment (ericholscher) [inline docs/user/analytics.rst:1]: If we're not using the `How To` in all the links to this section, why don't we just remove it from the title? It feels odd to only use it here..

File: docs/user/index.rst
Comment (ericholscher) [inline docs/user/index.rst:82]: I agree we need a feature overview page before this gets merged. Figuring out what's on the page seems a bit tricky if it doesn't include how to enable/use it..

File: docs/user/analytics.rst
Comment (benjaoming) [reply docs/user/analytics.rst:1]: This is revisiting an old decision. I followed this pattern :100: in all changes, so I don't think it's nice to have to go and undo this pattern now.

There is a note already about supporting a better title/nagivation title/intro in https://github.com/readthedocs/meta/discussions/74

See: https://github.com/readthedocs/readthedocs.org/pull/9620

File: docs/user/guides/administrators.rst
Comment (benjaoming) [reply docs/user/guides/administrators.rst:16]: See comment above.

File: docs/user/index.rst
Comment (benjaoming) [reply docs/user/index.rst:82]: The lack of a good feature description for Traffic Analysis is already noted. So I wanted to move forwards with the small refactors firstly before starting to write new contents.

But it seems I'm outnumbered now since you've also joined the other camp :joy:

Comment (ericholscher) [reply docs/user/index.rst:82]: I mean, this is actively making the docs worse for users to discover our features, so we should do the proper change.

Comment (benjaoming) [reply docs/user/index.rst:82]: We are in a process, each incremental step holds a direction towards Diátaxis. I don't agree that _when_ we re-label a 90% How To guide as a How To Guide, things are worsened. I think they improve. We note for the immediate future that we need a feature reference authored.

I recommend this: https://diataxis.fr/how-to-use-diataxis/

Especially "Work one step at a time" and "Just do something".

Comment (ericholscher) [reply docs/user/index.rst:82]: Sure, the relabeling isn't the issue, it's the fact that we no longer list a primary feature in our feature list.

Comment (benjaoming) [reply docs/user/index.rst:82]: Of course I want our features represented. I'm stressing that there is no current content for such a feature page because the existing content was a How-To. Therefore, I'm writing down imminent subsequent steps.

The plan was to start by making small steps to move contents into their right places. I think that's what I'm doing with this PR. I have no problem at all with creating a feature page, too. But then we are changing the original plan.

Comment (ericholscher) [reply docs/user/index.rst:82]: The whole plan is to not have a plan :) A goal also needs to be working incrementally and not making the docs worse in the meantime. Instead of debating this, you should just write the feature page, it would take the same amount of time..

**Comment by benjaoming** [comment]
I think this fits quite well into the "Main Features" page which is the target of a later refinement.

So by adding the feature here, we can ensure that it's duly processed in a later iteration.

**Comment by benjaoming** [comment]
I think this fits quite well into the "Main Features" page which is the target of a later refinement.

So by adding the feature here, we can ensure that it's duly processed in a later iteration.

File: docs/user/analytics.rst
Comment (ericholscher) [inline docs/user/analytics.rst:1]: Seems this is a How To guide, but not in the `guides` directory? 

Seems we should break this out into a How To for Google analytics, and the overall feature page which isn't a how to?

File: docs/user/features.rst
Comment (ericholscher) [inline docs/user/features.rst:78]: I would probably highlight both search & traffic analytics here?

File: docs/user/analytics.rst
Comment (benjaoming) [reply docs/user/analytics.rst:1]: Yes, this was on purpose though. I was thinking that the majority of the existing content (already index by Google) is a How-to. We have all the work with renaming files/folders and creating redirects postponed for later.

I agree that Google Analytics should be split into a separate guide. I'm going to ask some questions on Slack about its contents in this case.

**Comment by humitos** [comment]
@benjaoming can we avoid using force-push? In particular, after getting reviews from PRs. Otherwise, it resets my review and forces me to see all the differences again instead of keeping the files already reviewed collapsed and showing _only the new differences_ since my last view.

**Comment by benjaoming** [comment]
@humitos the force-push here happened because I merged in `main` by mistake.

File: docs/user/analytics.rst
Comment (ericholscher) [reply docs/user/analytics.rst:1]: This should be noted in the file somewhere then, but I guess the title & being in a guide index is a reasonably good start.

File: docs/user/features.rst
Comment (ericholscher) [reply docs/user/features.rst:78]: I also feel like we're trying to refactor away from this page, so adding more content to it feels wrong. I think it's probably better to draft new pages in our feature list for this, but I guess having them here before refactoring them later is OK?

**Review by ericholscher — COMMENTED** [review]
This is close 👍

**Comment by benjaoming** [comment]
re: @ericholscher 

> I think it's probably better to draft new pages in our feature list for this, but I guess having them here before refactoring them later is OK?

I need a bit more guidance here. What sort of feature page do you mean? How should it be included in the navigation? The strategy for the 4 iterations has been to convert feature pages into explanation. We have talked about a structure for documenting features through some kind of pattern, but this was left out.

**Comment by ericholscher** [comment]
@benjaoming 

I thought we explicitly said we're building the features list in the sidebar for now, and then moving it to Reference?

Screenshot 2022-12-12 at 3 09 50 PM

That is what I'm describing, is that analytics "feature" page, which will be under the sidebar `Features` page for now, but moved to `Reference > Features` later.

**Comment by benjaoming** [comment]
Ah, I missed that in the task list, thanks :+1:

File: docs/user/reference/features.rst
Comment (ericholscher) [inline docs/user/reference/features.rst:2]: 🤩

File: docs/user/reference/analytics.rst
Comment (ericholscher) [inline docs/user/reference/analytics.rst:13]:
```suggestion
Read the Docs aggregates statistics about visits to your documentation.
This is mainly information about how often pages are viewed,
and which return a `404 Not Found` error code. 
```

File: docs/user/reference/analytics.rst
Comment (ericholscher) [inline docs/user/reference/analytics.rst:17]: 💯 benefit/value statement.

File: docs/user/reference/analytics.rst
Comment (ericholscher) [inline docs/user/reference/analytics.rst:29]: This is about in-doc search, not incoming search terms. Shows how little we've been doing a good job of documenting it.  😆

Comment (benjaoming) [reply docs/user/reference/analytics.rst:29]: :facepalm: 

I didn't continue writing for the exact reason of feeling like something was terribly wrong :D

Comment (benjaoming) [reply docs/user/reference/analytics.rst:29]: This has been clarified and a new how-to for search analytics has been created :+1:

File: docs/user/analytics.rst
Comment (benjaoming) [reply docs/user/analytics.rst:1]: Yeah, I think we can keep it here for now. I fear that adding another page explaining analytics is a bit too much right now, because we have to tie together the SEO guide and existing contents that haven't yet trickled down into their correct navigation places.

Made a note of this in Iteration 4 :+1:

File: docs/user/features.rst
Comment (benjaoming) [reply docs/user/features.rst]: Moved :)

File: docs/user/reference/features.rst
Comment (benjaoming) [reply docs/user/reference/features.rst:2]: Taking shape! It should maybe become `user/reference/features/index.rst` but that can always happen.

Comment (benjaoming) [reply docs/user/reference/features.rst:2]: We have an outstanding item to think a bit more about how features should be structured, so I wanted the simplest approach.

File: docs/user/guides/search-analytics.rst
Comment (benjaoming) [inline docs/user/guides/search-analytics.rst:20]: This was super annoying to have to write.

But the list has a hidden scrollbar in my browser... Very easy to miss this scroll bar :/

File: docs/user/analytics.rst
Comment (benjaoming) [inline docs/user/analytics.rst:1]:
```suggestion
How to use traffic analytics
```

Question: Should we turn our features into Proper Nouns? In this case, that would make the headline be: "How to use Traffic Analytics".

File: docs/user/guides/search-analytics.rst
Comment (benjaoming) [inline docs/user/guides/search-analytics.rst:1]:
```suggestion
How to use search analytics
```

File: docs/user/analytics.rst
Comment (benjaoming) [reply docs/user/analytics.rst:1]: Nah, maybe not... simply looking at the amount of work to refactor all the texts makes this unrealistic :)

File: docs/user/guides/administrators.rst
Comment (benjaoming) [inline docs/user/guides/administrators.rst:19]:
```suggestion
   Use traffic analytics </analytics>
```

File: docs/user/guides/developers.rst
Comment (benjaoming) [inline docs/user/guides/developers.rst:21]:
```suggestion
   Use traffic analytics </analytics>
```

File: docs/user/reference/analytics.rst
Comment (benjaoming) [inline docs/user/reference/analytics.rst:1]:
```suggestion
Analytics for search and traffic
```

File: docs/user/reference/analytics.rst
Comment (benjaoming) [inline docs/user/reference/analytics.rst:10]:
```suggestion
Traffic analytics
```

File: docs/user/guides/search-analytics.rst
Comment (ericholscher) [inline docs/user/guides/search-analytics.rst:27]:
```suggestion
The number of times the query has been searched is displayed as the **searches** number.
```

File: docs/user/guides/search-analytics.rst
Comment (ericholscher) [inline docs/user/guides/search-analytics.rst:30]:
```suggestion
* If you see a search term that doesn't have any results,
  you could apply that term in documentation articles or create new ones.
  This is a great way to understand missing gaps in your documentation. 
```

File: docs/user/guides/search-analytics.rst
Comment (ericholscher) [inline docs/user/guides/search-analytics.rst:35]:
```suggestion
* Repeat the search yourself and inspect the results to see if they are relevant.
   You can add keywords to various pages that you want to show up for searches on that page. 
```

File: docs/user/guides/search-analytics.rst
Comment (ericholscher) [inline docs/user/guides/search-analytics.rst:1]: This is a much better "how to use" article, because it gives a lot of context and ideas for how to analyze the data. Veering slightly into explanation, but I think it's actually a great how to.

**Review by ericholscher — APPROVED** [review]
Looks great!

File: docs/user/guides/search-analytics.rst
Comment (benjaoming) [reply docs/user/guides/search-analytics.rst:1]: True that it tries to do a better job at giving _practical explanation_ to the use of what's visible on a particular page... but the explanation is centered around performing a specific task (i.e. the how-to title), so I hope it doesn't break our concept of the 4 axis.

@humitos you mentioned on another very "dull" step-by-step how-to that it barely provided value. The example was really good. I quoted your response [here](https://github.com/readthedocs/readthedocs.org/pull/9785#issuecomment-1361808732). For cases like this, does it make sense that we ask ourselves questions like:

* _Can this how-to contain recommendations and practical advice without breaking the how-to format?_
* _Can this how-to be expanded with relevant troubleshooting?_
* _Worst-case: Is this how-to describing a task that's so trivial and self-evident that we might as well remove it?_

File: docs/user/guides/search-analytics.rst
Comment (ericholscher) [reply docs/user/guides/search-analytics.rst]: exact query.

File: docs/user/guides/search-analytics.rst
Comment (ericholscher) [reply docs/user/guides/search-analytics.rst:1]: > * _Can this how-to contain recommendations and practical advice without breaking the how-to format?_
> * _Can this how-to be expanded with relevant troubleshooting?_
> * _Worst-case: Is this how-to describing a task that's so trivial and self-evident that we might as well remove it?_

I think these are great questions to ask. In particular, if we're representing something trivial, I think a small video/gif would likely do a lot more good for users. Perhaps something to try in the future. You can embed mp4/webp/etc. in webpages now as a gif really easily, and much smaller: https://www.smashingmagazine.com/2018/11/gif-to-video/#replace-animated-gifs-with-video-in-the-browser

Comment (benjaoming) [reply docs/user/guides/search-analytics.rst:1]: All added to the Style-guide PR :+1: #9840

**Pull request merged**
